src/effects/Skid.js

```diff
@@ -0,0 +1,42 @@
+import * as THREE from 'three'
+import { useRef } from 'react'
+import { useFrame } from '@react-three/fiber'
+import { useStore } from '../utils/store'
+
+const o = new THREE.Object3D()
+
+const skidLength = 200
+const skidSize = 0.4
+
+export function Skid({ opacity = 0.8 }) {
+  let index = 0
+  const { wheels } = useStore((state) => state.raycast)
+  const skid = useRef()
+
+  function setItemAt(ref, obj, i) {
+    o.position.set(obj.position.x, obj.position.y, obj.position.z)
+    o.rotation.set(-Math.PI / 2, 0, Math.random())
+    o.scale.setScalar(1)
+    o.updateMatrix()
+    ref.current.setMatrixAt(i, o.matrix)
+    ref.current.instanceMatrix.needsUpdate = true
+  }
+
+  useFrame(() => {
+    const { controls, sliding } = useStore.getState()
+
+    if (controls.brake && sliding) {
+      // Set new skid
+      setItemAt(skid, wheels[2].current, index++)
+      setItemAt(skid, wheels[3].current, index++)
+      if (index === skidLength) index = 0
```

Comment:
just trying to understand this (v new to three 😄 ): this means that the `Skid` consists of a max of 200 instances of a plane and you constantly move the trailing instances to the newest brake position if you've reached 200?